src/Numerics/Mesh/Topologies.jl

```diff
@@ -735,7 +784,190 @@ function StackedBrickTopology(
     nb = length(bndytoelem)
 
     T = eltype(basetopo.elemtocoord)
-
+    #----setting up DSS----------
+    if basetopo.elemtouvert isa Array
+        DA = ClimateMachine.array_type()
```

Comment:
At the moment we copy it over to the `DeviceArray` in grid.

Reply:
revised.